packages/playwright/src/runner/tasks.ts

```diff
@@ -161,15 +164,20 @@ function createPluginBeginTask(plugin: TestRunnerPluginRegistration): Task<TestR
   };
 }
 
-function createGlobalSetupTask(): Task<TestRun> {
+function createGlobalSetupTask(index: number, length: number): Task<TestRun> {
```

Comment:
I find the proposed behavior reasonable, so not really sure I understand the TODO in the description.

Reply:
I'd like to understand what the semantics here is. Let's say I have a database and an auth token. My database requires setup and teardown, my auth token only requires setup. Would there be a difference between the following configurations:

```js
globalSetup: [dbSetup.ts, authSetup.ts],
globalTeardown: dbTeardown.ts,
```
and

```
globalSetup: [authSetup.ts, dbSetup.ts],
globalTeardown: dbTeardown.ts,
```

This `index` parity confuses me and makes me think there is a difference.

Reply:
The difference is in execution ordering. First executes `dbSetup, authSetup ; dbTeardown`, second executes `authSetup, dbSetup ; dbTeardown`. In your usecase, this probably doesn't make a difference.